### PR TITLE
fix(redux) oidc default props

### PR DESCRIPTION
**Pull request by julienbirgand**
I made this pull request to fix this issue: #29 

I transformed Oidc to a class component to get rid of recompose's lifecycle method so there is no issue with defaultProps.

### CODE REVIEW

File: packages/redux/src/Oidc.spec.js [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
+
+// I used __esModule to mock a module with default and named exports
+// cf: https://github.com/facebook/jest/issues/5579#issuecomment-397406174
+jest.mock('./authenticationService', () => ({
+  __esModule: true,
+  default: jest.fn(),
+  getUserManager: jest.fn(),
+}));
+jest.mock('redux-oidc', () => ({
+  OidcProvider: jest.fn(),
```
Comment (guillaumechervetaxa) [inline packages/redux/src/Oidc.spec.js]: Thank you very much Julien, very nice PR.

Do you know this article below ?
https://medium.com/javascript-scene/mocking-is-a-code-smell-944a70c90a6a

I think it is possible to remove the mock and make your code more flexible and testable :)
Your welcome at my desk for some ideas !

**Comment by julienbirgand** [comment]
Link to recompose issue: https://github.com/acdlite/recompose/issues/502

File: packages/redux/src/Oidc.spec.js (hunk 1 above)
Comment (julienbirgand) [reply packages/redux/src/Oidc.spec.js]: I made some changes. Using a class component seemed to be a problem, so I used defaultProps enhancer from recompose. I still had to mock loadUser to check if it is called when the component mounts.

**Comment by guillaumechervetaxa** [comment]
your branch has effectively disapear :/

**Comment by youf-olivier** [comment]
> your branch has effectively disapear :/

julienbirgand:fix/redux-oidc-default-props => IMO The branch is in his own repo

**Comment by julienbirgand** [comment]
Maybe I should create a new PR?

**Pull request merged**